crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -5,26 +5,31 @@ use crate::ssa::{
 };
 use acvm::acir::native_types::Witness;
 use iter_extended::vecmap;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
 
-// maps memory address to expression
 #[derive(Default)]
-pub struct MemoryMap {
-    inner: HashMap<u32, InternalVar>,
+pub struct AcirMem {
+    // maps memory address to InternalVar
+    memory_map: BTreeMap<ArrayId, BTreeMap<u32, InternalVar>>,
```

Comment:
should this field be named `memory_maps`?

Reply:
Or perhaps `array_heaps` or similar to allude that it is the memory for each array.